components/rank/index.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from 'styled-components';
+import byteSize from 'byte-size';
+
+import { ArrowDown, ArrowUp } from '@components/icons';
+
+import device from '@utils/media_queries';
+import type { Data } from '@utils/dataType';
+
+const RankWrapper = styled.div`
+	font-size: 0.75rem;
+
+	@media ${device.tablet} {
+		font-size: 1rem;
+
+		& th {
+			&:first-child {
+				padding-left: 2rem;
+			}
+
+			&:last-child {
+				padding-right: 2rem;
+			}
+		}
+
+		& td {
+			&:first-child > span {
+				padding-left: 2rem;
+			}
+
+			&:last-child > span {
+				padding-right: 2rem;
+			}
+		}
+	}
+
+	& > table {
+		width: 100%;
+		border-collapse: collapse;
+	}
+
+	& th {
+		text-align: left;
+		padding-top: 1.75rem;
+		padding-bottom: 1.75rem;
+		background: #121212;
+		position: sticky;
+		top: 0;
+
+		&:first-child {
+			padding-left: 1rem;
+		}
+
+		&:last-child {
+			padding-right: 1rem;
+		}
+	}
+
+	& td {
+		&:first-child > span {
+			border-radius: 8px 0 0 8px;
+			padding-left: 1rem;
+		}
+
+		&:last-child > span {
+			border-radius: 0 8px 8px 0;
+			padding-right: 1rem;
+		}
+
+		& > span {
+			display: block;
+			background: #2c2c2c;
+			padding-top: 1rem;
+			padding-bottom: 1rem;
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+const formatWalletAddress = (wallet: string): string => {
+	const first5Chars = wallet.slice(0, 5);
+	const last5Chars = wallet.slice(-5);
+
+	return `${first5Chars}...${last5Chars}`;
+};
```

Comment:
I really like the way these variable names describe the string manipulation being used here, very helpful.

I do think all of these functions should be moved into the utils folder somewhere though. No need to pollute the components with all of this logic.

Reply:
Fixed in https://github.com/ardriveapp/inferno-leaderboard/commit/1f3a4d84239961d6c2d893dd393b8b23424d3b7b